diff -ruN fbv-0.99.orig/ChangeLog fbv-0.99/ChangeLog
--- fbv-0.99.orig/ChangeLog	Sun Aug 24 22:25:56 2003
+++ fbv-0.99/ChangeLog	Sun Oct 19 16:39:46 2003
@@ -1,3 +1,6 @@
+	2003-10-19	Pawe³ ¯urowski	pzurowski@post.pl
+	* added --nocenter option: image won't be in center of the sceen. Usefull with -c and -i options. Based on contributed patch from fbgetty.
+
 0.99    2003-08-24      Mateusz Golicz mtg@elsat.net.pl
       * fixes in transparent and interlaced GIF & PNG support
       * support for using the alpha channel
Binary files fbv-0.99.orig/fbv and fbv-0.99/fbv differ
diff -ruN fbv-0.99.orig/fbv.1 fbv-0.99/fbv.1
--- fbv-0.99.orig/fbv.1	Sun Aug 24 22:21:07 2003
+++ fbv-0.99/fbv.1	Sun Oct 19 16:39:46 2003
@@ -34,6 +34,9 @@
 .BR \fB--noinfo\fP , \fB-i\fP
 Do not show image information
 .TP
+.BR \fB--nocenter\fP , \fB-i\fP
+Do not center image on the screen
+.TP
 .BR \fB--stretch\fP , \fB-f\fP
 Strech (using simple resize) the image to fit onto screen if necessary
 .TP
diff -ruN fbv-0.99.orig/main.c fbv-0.99/main.c
--- fbv-0.99.orig/main.c	Sun Aug 24 22:26:57 2003
+++ fbv-0.99/main.c	Sun Oct 19 16:43:24 2003
@@ -38,7 +38,7 @@
 extern unsigned char * color_average_resize(unsigned char * orgin,int ox,int oy,int dx,int dy);
 
 
-int clear=1,delay=0,hide=1,dispinfo=1,allowstrech=0,opt_alpha = 0;
+int clear=1,delay=0,hide=1,dispinfo=1,cancenter=1,allowstrech=0,opt_alpha = 0;
 
 struct formathandler *fh_root=NULL;
 
@@ -167,8 +167,8 @@
 	        x=imx; y=imy;
 	    }
 	    
-	    if(x<xs) xpos=(xs-x)/2; else xpos=0;
-	    if(y<ys) ypos=(ys-y)/2; else ypos=0;
+	    if((x<xs) && cancenter) xpos=(xs-x)/2; else xpos=0;
+	    if((y<ys) && cancenter) ypos=(ys-y)/2; else ypos=0;
 	    xdelta=0; ydelta=0;
 
 	    xstep=min(max(x/20,1),xs);
@@ -242,6 +242,7 @@
 	       " --noclear 	| -c : Do not clear the screen before/after displaying image\n"
 	       " --unhide 	| -u : Do not hide/show the cursor before/after displaying image\n"
 	       " --noinfo 	| -i : Supress image information\n"
+	       " --nocenter 	| -e : Do not center image on the screen (usefull with -c and -i)\n"
 	       " --stretch	| -f : Strech (using a simple resizing routine) the image to fit onto screen if necessary\n"
 	       " --colorstretch | -k : Strech (using a 'color average' resizing routine) the image to fit onto screen if necessary\n"
            " --delay 	| -s <delay> slideshow, wait 'delay' tenths of a second before displaying each image\n\n"
@@ -263,6 +264,7 @@
     {"alpha", 	no_argument, 	0, 'a'},
 	{"unhide",  	no_argument, 	0, 'h'},
 	{"noinfo",  	no_argument, 	0, 'i'},
+	{"nocenter",  	no_argument, 	0, 'e'},
 	{"stretch", 	no_argument, 	0, 'f'},
 	{"colorstrech", no_argument, 	0, 'k'},
 	{"delay", 	required_argument, 0, 's'},
@@ -277,7 +279,7 @@
     {
 	for(;;)
 	{
-	    opt=getopt_long_only(argc,argv,"achukfis:",long_options,NULL);
+	    opt=getopt_long_only(argc,argv,"achukfies:",long_options,NULL);
 	    if(opt==EOF) break;
 	    switch(opt)
 	    {
@@ -287,6 +289,7 @@
 		case 'u': hide=0; break;
 		case 'h': help(argv[0]); break;
 		case 'i': dispinfo=0; break;
+		case 'e': cancenter=0; break;
 		case 'f': allowstrech=1; break;
 		case 'k': allowstrech=2; break;
 	    }
Binary files fbv-0.99.orig/main.o and fbv-0.99/main.o differ
